optional-goi-analysis/goi-calculations.R

```diff
@@ -133,11 +139,44 @@ if (!file.exists(opt$sce)){
   stop(paste(opt$sce, "does not exist."))
 }
 
-# Check that the outout directory exists
+# Check that the output directory exists
 if (!dir.exists(opt$output_directory)) {
   dir.create(opt$output_directory, recursive = TRUE)
 }
 
+# Define the output file paths
+output_mapped_genes <- file.path(
+  opt$output_directory,
+  paste0(opt$library_id, "_mapped_genes.tsv")
+)
+
+output_matrix <- file.path(
+  opt$output_directory,
+  paste0(opt$library_id, "_normalized_zscores.mtx")
+)
+
+output_annotation <- file.path(
+  opt$output_directory,
+  paste0(opt$library_id, "_heatmap_annotation.rds")
+)
+
+# Create a list of the expected output files
+output_files <- c(output_mapped_genes, output_matrix, output_annotation)
+
+# Check that any of the output files exist, only if the overwrite flag is false
+if (!opt$overwrite) {
+  if (any(file.exists(output_files))) {
+    existing_file <- output_files[file.exists(output_files)]
+    stop(
+      glue::glue(
+        "{existing_file} file exists, to overwrite please use the --overwrite flag
```

Comment:
If there are > 1 existing files, this will print out the same statement a few times. I might have separate printing here for the existing file indication and the `stop()` itself, to avoid overly-repetitive output.

```suggestion
glue::glue("{existing_file} file already exists.")
stop("To re-run analysis and overwrite all existing files please use the --overwrite flag.")
```

Reply:
I added a `print()` function to get this working -- now it prints out each of the files that already exist and the "To re-run analysis..." statement just once. 👍